Projects/Embark/Sources/EmbarkState+Tracking.swift

```diff
@@ -0,0 +1,55 @@
+import Foundation
+import hGraphQL
+import hCore
+
+public struct EmbarkTrackingEvent {
+    public var title: String
+    public var properties: [String:Any]
+    
+    func send() {
+        Self.trackingHandler(self)
+    }
+}
+
+public extension EmbarkTrackingEvent {
+    static var trackingHandler: (_ Event: EmbarkTrackingEvent) -> Void = { _ in }
+}
+
+internal extension GraphQL.EmbarkExternalRedirectLocation {
+    func trackingEvent(store: EmbarkStore) -> EmbarkTrackingEvent {
+        var properties = store.getAllValues()
+        properties["redirectLocation"] = self.rawValue
+        return EmbarkTrackingEvent(title: "External Redirect", properties: properties)
+    }
+}
+
+internal extension EmbarkPassage.Track {
+    func trackingEvent(store: EmbarkStore) -> EmbarkTrackingEvent {
+        return .init(title: self.eventName, properties: properties(store: store))
+    }
+    
+    private func properties(store: EmbarkStore) -> [String:Any] {
+        var properties = Dictionary<String,Any>()
+        if includeAllKeys {
```

Comment:
it's been decided that this is not something we wanna support going forward, due to there being unwanted data sent to Mixpanel, so we can remove this for now.

Reply:
But we should still send the entire dictionary on external redirect ?